bot/exts/moderation/silence.py

```diff
@@ -92,88 +113,230 @@ async def _async_init(self) -> None:
+            else:
+                await self._force_voice_sync(channel)
+
+        await self._schedule_unsilence(ctx, channel, duration)
 
         if duration is None:
-            self.notifier.add_channel(ctx.channel)
+            self.notifier.add_channel(channel)
             log.info(f"Silenced {channel_info} indefinitely.")
-            await ctx.send(MSG_SILENCE_PERMANENT)
+            await self.send_message(MSG_SILENCE_PERMANENT, ctx.channel, channel, alert_target=True)
+
         else:
             log.info(f"Silenced {channel_info} for {duration} minute(s).")
-            await ctx.send(MSG_SILENCE_SUCCESS.format(duration=duration))
+            formatted_message = MSG_SILENCE_SUCCESS.format(duration=duration)
+            await self.send_message(formatted_message, ctx.channel, channel, alert_target=True)
 
     @commands.command(aliases=("unhush",))
-    async def unsilence(self, ctx: Context) -> None:
+    async def unsilence(self, ctx: Context, *, channel: TextOrVoiceChannel = None) -> None:
         """
-        Unsilence the current channel.
+        Unsilence the given channel if given, else the current one.
 
         If the channel was silenced indefinitely, notifications for the channel will stop.
         """
         await self._init_task
-        log.debug(f"Unsilencing channel #{ctx.channel} from {ctx.author}'s command.")
-        await self._unsilence_wrapper(ctx.channel)
+        if channel is None:
+            channel = ctx.channel
+        log.debug(f"Unsilencing channel #{channel} from {ctx.author}'s command.")
+        await self._unsilence_wrapper(channel, ctx)
 
     @lock_arg(LOCK_NAMESPACE, "channel", raise_error=True)
-    async def _unsilence_wrapper(self, channel: TextChannel) -> None:
-        """Unsilence `channel` and send a success/failure message."""
+    async def _unsilence_wrapper(self, channel: TextOrVoiceChannel, ctx: Optional[Context] = None) -> None:
+        """
+        Unsilence `channel` and send a success/failure message to ctx.channel.
+
+        If ctx is None or not passed, `channel` is used in its place.
+        If `channel` and ctx.channel are the same, only one message is sent.
+        """
+        msg_channel = channel
+        if ctx is not None:
+            msg_channel = ctx.channel
+
         if not await self._unsilence(channel):
-            overwrite = channel.overwrites_for(self._everyone_role)
-            if overwrite.send_messages is False or overwrite.add_reactions is False:
-                await channel.send(MSG_UNSILENCE_MANUAL)
+            if isinstance(channel, VoiceChannel):
+                overwrite = channel.overwrites_for(self._verified_voice_role)
+                manual = overwrite.speak is False
             else:
-                await channel.send(MSG_UNSILENCE_FAIL)
+                overwrite = channel.overwrites_for(self._everyone_role)
+                manual = overwrite.send_messages is False or overwrite.add_reactions is False
```

Comment:
The `manual` name didn't really explain the var's purpose for me, not sure what would be the most fitting, maybe something like `has_manual_overwrites`

Reply:
I'm not sure what to call it either. Manual implies there's an automatic overwrite, but that's not really the case. A channel either follows the category's overwritten, or it is manually overwritten. I dropped the overwrite part for `has_channel_overwrite`, but I'm happy to change it again. 52bfbf31